enhancements/network/allow-mtu-and-overlay-port-changes.md

```diff
@@ -0,0 +1,165 @@
+9. Wait a grace period. This grace period is based on
+   `CHANGE_MTU_INIT_TS` rather than the time of finalization of the
+   previous step.
+10. In the case of OVN Kubernetes proceed to change the MTU of the interfaces
+   ovn-k8s-mp0, br-local, and ovn-k8s-gw0 sequentially without any delay in
+   between.
+
+   In the case of OpenShiftSDN proceed to change the MTU of the interfaces
+   vxlansys.
+
+13. Uncordon the node
+
+14. Annotate itself with `openshift.io/cno-succesuflly-changed-mtu` and exit
+    with code 0.
+
+When increasing the MTU the same process applies, except steps 10, 8, and 6 are done in reverse order.
+
+Changing the overlay ports is a similar procedure, we will depoy a daemonset
+where each pod will:
+
+1. Verify that the port is not being used at the moment by anything else and
+   add to itself the annotation `openshift.io/cno-able-to-change-port`
+2. Wait until the timestamp defined in `CHANGE_PORT_INIT_TS`
+3. Verify that every pod of that daemonset is running and has the annotation
+   `openshift.io/cno-able-to-change-port` if any of the pods is not running, or
+   is missing that annotation the pod will abort the procedure.
+4. Cordon the node where its running.
+5. Wait a grace period. This grace period is based on
+   `CHANGE_PORT_INIT_TS` rather than the time of finalization of the
+   previous step.
+6. Actually change the port:
+   For OpenShift SDN we'll delete the port:
+
+   `ovs-vsctl --if-exists del-port br0 vxlan0`
+
+   And create it again with no delay at all:
+
+   ```
+   ovs-vsctl add-port br0 vxlan0 \
+     -- set Interface vxlan0 ofport_request=1 \
+     type=vxlan options:remote_ip="flow" \
+     options:key="flow" \
+     options:dst_port=<port number>
+   ```
+
+   In the case of OVN Kubernetes we'll do the same procedure. If we need to
+   change the vxlan port we'll gather its configuration from ovs-vsctl, delete
+   it, and recreate it with the same parameters.
+
+   And if we need to change the geneve port we'll do the same thing with it.
+
+7. In the case of OpenShift SDN each node will uncordon iself.
+8. EAch pod will annotate the node where it's running with
+   `openshift.io/cno-succesuflly-changed-mtu` and exit
+
+Once the daemonsets are finished, for OVN Kubernetes the CNO will verify that
+the configmap ovnkube-config is synchronized, and will force a rollout of the
+ovnkube-master daemonset. After that it will also force a rollout of the
+ovnkube-node daemonset.
+
+If any of the nodes fails during the procedure we expect the administrator to
+manually reboot the nodes. We could reboot automatically, but given the risks
+involved, and that this change is a very uncommon intervention, we consider
+safer making the human responsible for it.
+
+### Test Plan
+For MTU changes start a TCP server and client in two different nodes
+communicating continuosuly. Decrease the MTU and increase it once its finished.
+The TCP connection must be alive after both changes.
```

Comment:
would also be good to have a process repeatedly opening new connections, transferring a bit, closing, and looping.

Also, maybe explicitly do something involving TLS and large certificates, since that's a common MTU failure mode...

Reply:
I've decided to rewrite the whole test plan with your feedback, I think it now addresses all your thoughts.